test/shared/src/main/scala/zio/test/Spec.scala

```diff
@@ -338,7 +348,7 @@ final case class Spec[-R, +E, +L, +T](caseValue: SpecCase[R, E, L, T, Spec[R, E,
         case SuiteCase(label, specs, exec) =>
           specs.flatMap(ZIO.foreach(_)(loop(r))).map(z => Spec.suite(label, ZIO.succeed(z.toVector), exec))
         case TestCase(label, test) =>
-          test.map(t => Spec.test(label, ZIO.succeed(t)))
+          test.run.map(exit => Spec.test(label, ZIO.done(exit)))
```

Comment:
Could this affect any of the other `provide*` variants?

Reply:
This only impacts `shared` variants and all `shared` variants are implemented in terms of `provideManagedShared` so we just need to fix it here.